cmd/nerdctl/system_prune.go

```diff
@@ -99,5 +103,70 @@ func systemPruneAction(cmd *cobra.Command, args []string) error {
 			return err
 		}
 	}
-	return imagePrune(cmd, client, ctx)
+	if err := imagePrune(cmd, client, ctx); err != nil {
+		return nil
+	}
+	prunedObjects, err := buildCachePrune(ctx, cmd, all)
+	if err != nil {
+		return nil
+	}
+
+	if len(prunedObjects) > 0 {
+		fmt.Fprintln(cmd.OutOrStdout(), "Deleted build cache objects:")
+		for _, item := range prunedObjects {
+			fmt.Fprintln(cmd.OutOrStdout(), item.ID)
+		}
+	}
+
+	// TODO: print total reclaimed space
+
+	return nil
+}
+
+type cacheUsageInfo struct {
+	ID   string
+	Size int64
+}
+
+func buildCachePrune(ctx context.Context, cmd *cobra.Command, pruneAll bool) ([]cacheUsageInfo, error) {
+	buildkitHost, err := getBuildkitHost(cmd)
+	if err != nil {
+		return nil, err
+	}
+	opts := []buildkitclient.ClientOpt{buildkitclient.WithFailFast()}
+	client, err := buildkitclient.New(ctx, buildkitHost, opts)
+	if err != nil {
+		return nil, err
+	}
+	usageCh := make(chan buildkitclient.UsageInfo)
+	var pruneOpts []buildkitclient.PruneOption
+	if pruneAll {
+		pruneOpts = append(pruneOpts, buildkitclient.PruneAll)
+	}
+	var usageInfos []buildkitclient.UsageInfo
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go func() {
+		defer close(usageCh)
+		defer wg.Done()
+		for item := range usageCh {
+			usageInfos = append(usageInfos, item)
+		}
+	}()
+	err = client.Prune(ctx, usageCh, pruneOpts...)
+	if err != nil {
+		return nil, nil
+	}
+
+	wg.Wait()
```

Comment:
It seems that `nerdctl system prune` hangs. 
`usageCh` needs to be closed after `client.Prune` returned because it doesn't close that channel.